examples/vision/sample_size_estimate.py

```diff
@@ -0,0 +1,545 @@
+"""
+Title: Estimating required sample size for model training
+Author: [JacoVerster](https://twitter.com/JacoVerster)
+Date created: 2021/05/20
+Last modified: 2021/05/21
+Description: Estimate the number of samples required to reach a desired model accuracy.
+"""
+
+"""
+# Introduction
+In most real-world use cases, the image data required to train a deep learning model is
+limited, this is especially true in the medical imaging domain where dataset creation is
+costly. The first question that usually comes up is "how many images will we need to
+train a good enough machine learning model".
+
+Statistical approaches can be used to try and calculate the size of an optimum training
+set, but this is difficult. In many cases a small set of data is available and then it is
+useful to consider the “behaviour” of the model when subjected to varying training sample
+sizes. A model-based sample size determination method can be used to estimate the optimum
+number of images needed to arrive at a scientifically valid sample size that would give
+the required model performance.
+
+A systematic review of [Sample-Size Determination Methodologies](https://bit.ly/3f53LSs)
+by Balki et al. provides examples of several sample-size determination methods. In this
+example, a balanced subsampling scheme is used to determine the optimum sample size for
+our model. This is done by selecting a random subsample consisting of Y number of images
+and training the model using the subsample. The model is then evaluated on an independent
+test set. This process is repeated N times for each subsample with replacement to allow
+for the construction of a mean and confidence interval for the observed performance.
+
+This example requires TensorFlow 2.4 or higher.
+"""
+
+"""
+# Setup
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+import tensorflow_datasets as tfds
+from tensorflow.keras import layers
+from tensorflow.keras.layers.experimental import preprocessing
+
+# Define seed and fixed variables
+seed = 42
+tf.random.set_seed(seed)
+np.random.seed(seed)
+AUTO = tf.data.AUTOTUNE
+
+"""
+# Load TensorFlow dataset
+
+We'll be using the [TF Flowers dataset](https://bit.ly/34FQxWc).
```

Comment:
Use the direct link rather than bitly, https://www.tensorflow.org/datasets/catalog/tf_flowers